src/QMCHamiltonians/Listener.hpp

```diff
@@ -0,0 +1,66 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2022 QMCPACK developers.
+//
+// File developed by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//
+// File created by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef QMCPLUSPLUS_LISTENER_HPP
+#define QMCPLUSPLUS_LISTENER_HPP
+/** \file
+ *  Listener types that allow Estimators to register with QMCHamiltonian to have "trace" values from
+ *  operators reported.
+ *  This is aims to be lightweight and minimal but still use much of the trace manager based
+ *  implementation of observables that rely on per particle Hamiltonian values.
+ */
+
+#include <functional>
+
+#include "type_traits/template_types.hpp"
+
+#include "OhmmsPETE/OhmmsMatrix.h"
+#include "OhmmsPETE/OhmmsVector.h"
+
+namespace qmcplusplus
+{
+
+/** An object of this type is a listener expecting a callback to the report function  with a vector of values,
+ *  the convention being 1 per particle, called once per walker per component.
+ *  The name is primarily for debugging purposes, if have decided against having the QMCHamiltonian use it to estable
+ *  routing. Instead the register functions in the QMCHamiltonian are specfic to the sort of value that the listener wants
+ *  to listen to.
+ */
+template<typename T>
+class ListenerVector
```

Comment:
Are there going to be other types passed to the report function? In other words, is there going to be a class named "ListenerScalar"?   Seeing the name ListenerVector, I immediately think of a vector of Listeners, but it is not.  Reading the docs, it looks like the "Vector" part means it expects a vector input.

Reply:
Your point is good.  I did have a ListenerVar and ListenerMatrix but they haven't proved to be useful yet so I left them out of this PR. Maybe ListenerToVector would be better, naming these sorts of things is tricky.